src/borg/_hashindex.c

```diff
@@ -447,19 +461,25 @@ hashindex_set(HashIndex *index, const void *key, const void *value)
         memcpy(entry_to_insert, key, index->key_size);
         memcpy(entry_to_insert + index->key_size, value, index->value_size);
         bucket_ptr = BUCKET_ADDR(index, idx);
-        /* collisions += offset;  //TRACE_PERF */
+        if (DEBUG) {
+            collisions += offset;
+        }
         while(!BUCKET_IS_EMPTY(index, idx) && !BUCKET_IS_DELETED(index, idx)) {
             /* we have a collision */
             other_offset = distance(
                 num_buckets, idx, hashindex_index(index, bucket_ptr));
-              /* collisions++;  //TRACE_PERF */
+            if (DEBUG) {
+                collisions++;
+            }
             if(other_offset < offset) {
                 /* Swap the bucket at idx with the current entry_to_insert.
                    This is the gist of robin-hood hashing, we rob from the key with the
                    lower distance to its optimal address by swapping places with it. */
                 memswap(bucket_ptr, entry_to_insert, tmp_entry, entry_size);
                 offset = other_offset;
-              /* swaps++;  //TRACE_PERF */
+                if (DEBUG) {
+                    swaps++;
+                }
```

Comment:
the whole "if debug increment x" could be also INC(x) macro which expands via precompiler to code or nothing, depending on DEBUG value. would save some code lines maybe.